node/core/provisioner/src/disputes/prioritized_selection/mod.rs

```diff
@@ -364,10 +364,20 @@ fn is_vote_worth_to_keep(
 	dispute_statement: DisputeStatement,
 	onchain_state: &DisputeState,
 ) -> bool {
-	let offchain_vote = match dispute_statement {
-		DisputeStatement::Valid(_) => true,
-		DisputeStatement::Invalid(_) => false,
+	let (offchain_vote, valid_kind) = match dispute_statement {
+		DisputeStatement::Valid(kind) => (true, Some(kind)),
+		DisputeStatement::Invalid(_) => (false, None),
 	};
+	// We want to keep all backing votes. This maximizes the number of backers
+	// punished when misbehaving.
+	if let Some(kind) = valid_kind {
+		match kind {
+			ValidDisputeStatementKind::BackingValid(_) => return true,
+			ValidDisputeStatementKind::BackingSeconded(_) => return true,
+			_ => (),
+		}
+	}
```

Comment:
nit: Your version is totally fine, but I think you can also do something like this 🙂 (haven't checked it):

```suggestion
    if matches!(valid_kind, Some(ValidDisputeStatementKind::BackingValid(_) | ValidDisputeStatementKind::BackingSeconded(_)) {
		return true
	}
```